pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -654,7 +654,10 @@ let
 
       MODULE_COMPRESS    = yes;
       MODULE_COMPRESS_XZ = yes;
-      KERNEL_XZ          = yes;
+
+      # use zstd for kernel compression if newer than 5.9, else xz.
+      KERNEL_XZ          = { optional = true; tristate = whenOlder "5.9" "y";};
+      KERNEL_ZSTD        = { optional = true; tristate = whenAtLeast "5.9" "y";};
```

Comment:
I don't think these need to be `optional`? The version conditions should be enough to determine if it's available, and if it's not available in spite of the version theoretically supporting it then there's something funky going on and we should probably error out.

Reply:
Yup, works without; much simpler.